src/System.Management.Automation/engine/CommandCompletion/ExtensibleCompletion.cs

```diff
@@ -102,7 +102,127 @@ IEnumerable<CompletionResult> CompleteArgument(
             string parameterName,
             string wordToComplete,
             CommandAst commandAst,
-            IDictionary fakeBoundParameters);
+            IDictionary fakeBoundParameters)
+        {
+            CommandName = commandName;
+            ParameterName = parameterName;
+            WordToComplete = wordToComplete;
+            CommandAst = commandAst;
+            FakeBoundParameters = fakeBoundParameters;
+            return ShouldComplete ? GetMatchingResults(wordToComplete) : [];
+        }
+
+        /// <summary>
+        /// Gets the name of the command that needs argument completion.
+        /// </summary>
+        protected static string? CommandName { get; private set; }
+
+        /// <summary>
+        /// Gets the name of the parameter that needs argument completion.
+        /// </summary>
+        protected static string? ParameterName { get; private set; }
+
+        /// <summary>
+        /// Gets the word being completed.
+        /// </summary>
+        protected static string? WordToComplete { get; private set; }
+
+        /// <summary>
+        /// Gets the command abstract syntax tree (AST).
+        /// </summary>
+        protected static CommandAst? CommandAst { get; private set; }
+
+        /// <summary>
+        /// Gets the fake bound parameters similar to $PSBoundParameters.
+        /// </summary>
+        protected static IDictionary? FakeBoundParameters { get; private set; }
```

Comment:
I guess here I wanted these to only be set in `CompleteArgument`. I made them static for this case but it might not be worth it